### ISSUE TITLE
[UoM] Feature request: Apply TimeFormat for labels of Number:Time items

### ISSUE THREAD

**Issue by cweitkamp** [eclipse/smarthome#5506]
Assuming we have an item of type "Number:Time" with seconds as source unit. We can display the value of the source unit with the pattern "[%d %unit%]". A conversion to minutes is possible with the pattern "[%.1f min]". A value of 80s will be displayed as "80 s" with the first pattern whereas it will be displayed as "1.3 min" with the second pattern.

It would be a great feature if it is possible to use a TimeFormat pattern (e.g. "[%1$tM min %1$tS s]") to display the value in a human readable and used format: "1 min 20 s".

**Comment by htreu** [comment]
Hi @cweitkamp, this would be nice indeed. Do you think you could provide a contribution for this?

**Comment by cweitkamp** [comment]
@htreu Yes of course but I am currently a little bit busy. What do you think is the right place to start? Method `QuantityType.format()`?
https://github.com/eclipse/smarthome/blob/b5839c8e835bcc8b519e5d86f61807adb109dae2/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/QuantityType.java#L225

**Comment by htreu** [comment]
Yes, the `format` method will be called every time a state description pattern should be applied. Its the correct place to do some pre calculations.
